build-scripts/MSVC.cmake

```diff
@@ -0,0 +1,71 @@
+if (NOT "$ENV{DevEnvDir}")
+    # Use Community Edition when not specified
+    set(ENV{DevEnvDir} "C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\Common7\\IDE\\")
+endif()
+if(CMAKE_GENERATOR MATCHES ^Ninja)
+    # Ninja is provided by Microsoft but not in the Path
+    set(CMAKE_MAKE_PROGRAM $ENV{DevEnvDir}CommonExtensions\\Microsoft\\CMake\\Ninja\\ninja.exe CACHE PATH "")
+endif()
+
+# Path has changed, so this configure run will find cl.exe
+set(CMAKE_C_COMPILER   cl.exe)
+set(CMAKE_CXX_COMPILER ${CMAKE_C_COMPILER})
+set(CMAKE_CXX_FLAGS_INIT "\
+/MP /utf-8 /bigobj /permissive- /sdl- /FC /Gd /GS- /Gy /GF \
+/wd4068 /wd4146 /wd4819 /wd6237 /wd6319 /wd26444 /wd26451 /wd26495 /WX- /W1 \
+/TP /Zc:forScope /Zc:inline /Zc:wchar_t"
+)
+# /MP     # cl.exe build with multiple processes
+# /utf-8  # set source and execution character sets to UTF-8
+# /bigobj # increase # of sections in object files
+# /permissive- # to allow alternative operators ("and", "or", "not")
+# /sdl-   # disable additional security checks
+# /FC     # full path in compiler messages
+# /Gd     # __cdecl
+# /GS-    # disable buffer security checks
+# /Gy     # Enable Function-Level Linking
+# /GF     # Eliminate Duplicate Strings
```

Comment:
Idle commeentary: technically it's 'package static data in COMDATS' which enables `/OPT:ICF` to deduplicate it the same way that `/Gy` does it for functions.